tests/datadriven/test_reads.py

```diff
@@ -67,6 +71,25 @@ def getDataModelClass(self):
     def getProtocolClass(self):
         return protocol.ReadGroupSet
 
+    def testPrograms(self):
+        # test that program info is set correctly
+        readGroupSet = self._gaObject
+        for readGroup in readGroupSet.getReadGroups():
+            readGroupInfo = self._readGroupInfos[readGroup.getSamFilePath()]
+            gaPrograms = readGroup.getPrograms()
+            htslibPrograms = readGroupInfo.programs
```

Comment:
Gah, always forget about that.  It would probably be good to implement a `zipSafe` or something which throws an error if the lengths are not the same.